cowprotocol/cowswap
# [EthFlow]#1294 handle failed txs

# Summary

Part of #1294 

Handle creation transaction failure

<img width="815" alt="Screenshot 2022-12-20 at 14 43 43" src="https://user-images.githubusercontent.com/43217/208699072-e3782d46-02be-469b-b9b7-5858c3594e5e.png">

![Screenshot 2022-12-20 at 15 27 19](https://user-images.githubusercontent.com/43217/208703464-142bca0d-d87e-4388-ae14-4cd9e722972a.png)


- New order state `invalid` - this will be a new state returned by the backend
- Updated stepper to display cancellation failure

## Notes
1. Not tested with transaction cancellation/speed up. May or may not work (more likely won't)
2. Cancellation transaction failure not yet handled

  # To Test

1. Place an ETH sell order for 1 ETH
* Tx will fail
* Stepper will be updated to show the status like the picture above
2. Place regular ETH sell orders
* Should work as usual

How do you fail a tx, you ask?
I've made a temporary change so every ETH flow order will always fail when value is exactly `1 ETH` in this PR :)
